tools/challenge-md-parser/solution-to-data.js

```diff
@@ -2,17 +2,49 @@ const visit = require('unist-util-visit');
 const { selectAll } = require('hast-util-select');
 
 const { sectionFilter } = require('./utils');
+const { createCodeGetter, defaultFile } = require('./challengeSeed-to-data');
+const { isEmpty } = require('lodash');
+
+const solutionRE = /(.+)-solution$/;
+
+function indexByKey(obj) {
+  return { [obj.key]: { ...obj } };
+}
 
 function createPlugin() {
   return function transformer(tree, file) {
     function visitor(node) {
       if (sectionFilter(node, 'solution')) {
-        const solutions = selectAll('code', node).map(
-          element => element.children[0].value
+        // fallback for single-file challenges
+        const rawSolutions = selectAll('code', node).map(element => ({
+          lang: element.properties.className[0].split('-')[1],
+          contents: element.children[0].value
+        }));
+
+        const solutionFiles = {};
+
+        const codeDivs = selectAll('div', node);
+        const solutionContainers = codeDivs.filter(({ properties: { id } }) =>
+          solutionRE.test(id)
         );
+        solutionContainers.forEach(
+          createCodeGetter('contents', solutionRE, solutionFiles)
+        );
+
+        const solutionsAsFiles = rawSolutions
+          .map(({ lang, contents }) => ({
+            ...defaultFile(lang),
+            contents
+          }))
+          .map(indexByKey);
+
+        const solutions = isEmpty(solutionFiles)
+          ? solutionsAsFiles
+          : [solutionFiles];
+
         file.data = {
           ...file.data,
-          solutions
+          solutions: solutions
```

Comment:
What was wrong with just `solutions` here?

Reply:
I think that slipped through when I removed `solutionFiles` from the schema. It should just be `solution`

Reply:
I've fixed it.  Once you give the final go-ahead, I'll clean up the history a bit and we can merge.

Reply:
...and I will remove the DEBUG challenge again.